app-modules/ai/src/Filament/Resources/AiAssistants/Pages/ManageAiAssistantAdditionalKnowledge.php

```diff
@@ -0,0 +1,216 @@
+                                        ->placeholder('Not parsed yet')
+                                        ->disabled()
+                                        ->visible($user->isSuperAdmin()),
+                                ]),
+                                fn (Repeater $repeater) => $repeater->simple(
+                                    TextInput::make('name')
+                                        ->disabled(),
+                                ),
+                            )
+                            ->addable(false)
+                            ->visible(fn (AiAssistant $record): bool => $record->files->isNotEmpty())
+                            ->deleteAction(
+                                fn (Action $action) => $action->requiresConfirmation()
+                                    ->modalHeading('Are you sure you want to delete this file?')
+                                    ->modalDescription('This file will be permanently removed from this Custom Advisor, and cannot be restored.')
+                            ),
+                        FileUpload::make('uploaded_files')
+                            ->hiddenLabel()
+                            ->multiple()
+                            ->reactive()
+                            ->maxFiles(fn (AiAssistant $record): int => 25 - $record->files->count())
+                            ->disabled(fn (AiAssistant $record): bool => $record->files->count() >= 25)
+                            ->acceptedFileTypes(config('ai.supported_file_types'))
+                            ->storeFiles(false)
+                            ->helperText(function (AiAssistant $record): string {
+                                if ($record->files->count() < 25) {
+                                    return 'You may upload a total of 25 files to this AI Assistant. Files must be less than 20MB.';
+                                }
+
+                                return "You've reached the maximum file upload limit of 25 for this Custom Advisor. Please delete a file if you wish to upload another.";
+                            })
+                            ->maxSize(20000)
+                            ->columnSpan(function (Get $get) {
+                                $files = $get('files');
+                                $firstFile = reset($files);
+
+                                if (! $firstFile || blank($firstFile['name'])) {
+                                    return 'full';
+                                }
+
+                                return 1;
+                            }),
+                    ]),
+            ]);
+    }
+
+    public function getRedirectUrl(): ?string
+    {
+        return $this->getUrl(['record' => $this->getRecord()]);
+    }
+
+    protected function handleRecordUpdate(Model $record, array $data): Model
+    {
+        $record->fill($data);
+
+        assert($record instanceof AiAssistant);
+
+        if (filled($data['uploaded_files'] ?? null)) {
```

Comment:
Rather than do all this here manually, are we about to use the Trait that was being used before, `HandlesFileUploads`?

It also looks like you handled removing calling the method in that Trait in `CreateAiAssistant` but not the edit page.

Also assess whether or not the Trait is needed in the create page at all now, if there is nothing in it being used that trait can also be removed.